### PR TITLE
Add ability to set Duo user status and query Duo admin name

### PR DESCRIPTION
**Pull request by kylekoza**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
There are several features in the Duo Admin API that are not supported by this integration.
This pull requests adds support for modifying a Duo user's status (e.g. active, disabled, or bypass) and 
also allows querying for a Duo admins information from their name (which is the field returned in several logs types)

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Demisto
- [ ] 5.0.0
- [ ] 5.5.0
- [X] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [X] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/DuoAdminApi/Integrations/DuoAdminApi/DuoAdminApi.py [hunk 1]
```diff
@@ -330,6 +330,27 @@ def delete_u2f_token(token_id):
...
+def set_user_status(username, status):
+    result = admin_api.get_users_by_name(username)
+    user_id = result[0]["user_id"]
+    res = admin_api.update_user(user_id, status=status)
+    demisto.results('User ' + username + 'was set to status ' + status)
+
+
+def get_duo_admin_by_name(name):
+    duo_administrators = admin_api.get_admins()
+    duo_administrator = next((admin for admin in duo_administrators if admin['name'] == name), None)
```
Comment (Bargenish) [inline Packs/DuoAdminApi/Integrations/DuoAdminApi/DuoAdminApi.py:342]: if it's possible to get an admin ID using the `get_user_id(username)` function, please change the API call to `get_admin` (see ref: https://github.com/duosecurity/duo_client_python/blob/master/duo_client/admin.py#L2496)

```suggestion
    admin_id = get_user_id(name)
    duo_administrator = admin_api.get_admin(admin_id)
```

File: Packs/DuoAdminApi/Integrations/DuoAdminApi/DuoAdminApi.py [hunk 2]
```diff
@@ -330,6 +330,27 @@ def delete_u2f_token(token_id):
...
+
+    entry = get_entry_for_object(
+        'Information about admin ' + name, duo_administrator, duo_administrator,
+        {
+            'DuoAdmin.AdminDetail(val.name && val.name==obj.name)':
+                {'details': duo_administrator}
```
Comment (Bargenish) [inline Packs/DuoAdminApi/Integrations/DuoAdminApi/DuoAdminApi.py:348]: Few notes:
1. You dont have a `DuoAdmin.AdminDetail.name` field in that case but ``DuoAdmin.AdminDetail.details.name``. either add it to the context as follows or fix the path.
```suggestion
            'DuoAdmin.AdminDetail(val.name && val.name==obj.name)':
                {
                'name': name,
                'details': duo_administrator
                }
```
2. What data does `duo_administrator` hold? please add the related fields to the command's outputs in the yml file. 
you can use `json_to_output` for that matter. You can see an example and explanation for this [here](https://xsoar.pan.dev/docs/integrations/context-and-outputs#integration-yml)

**Review by Bargenish — COMMENTED** [review]
@kylekoza Thanks for your contribution! 

- Please add revision version release notes. - Read about it [here](https://xsoar.pan.dev/docs/integrations/release-notes)
- Please add the new commands to the `Test Playbook` - Read about it [here](https://xsoar.pan.dev/docs/integrations/test-playbooks)
- Please add outputs for the `duoadmin-get-admin-by-name` command.

**Comment by Bargenish** [comment]
Hi there @kylekoza if you are aiming to contribute this pack please address the relevant comments so we can get this pack merged, if not please let us know as soon as possible, as we are allocating resources to handle this contribution process.

**Comment by kylekoza** [comment]
@Bargenish yes, I plan to work on this

**Comment by bakatzir** [comment]
@kylekoza is there anything we can do to speed up the comment fixes? would you want to set a short zoom for further explanation or are all the comments clear?

**Comment by bakatzir** [comment]
Closing, let's re open when you get back to it.

**Pull request closed without merging**